### DISCUSSION TITLE
Does exist a SQL to JooQ converter in Kotlin\Jooq code?

### DISCUSSION THREAD

**Discussion by nkamp** [jOOQ/jOOQ#17004] [Q&A] [answered]
I'm new to Jooq and Kotlin. I have SQL query which is working. 
I have to convert this into JooQ code. 

Does there exist which can convert SQL code into JooQ (Kotlin) Code? Or do I have to build the JooQ query on my own?

Here is my SQL query:
```
SELECT (CASE WHEN COUNT(*) > 0 THEN true
            ELSE false
            END) AS inconsistent
FROM tableA zz
    ,tableB za
    ,tableC zai
WHERE zz.zaak_identificatie = 1007
  AND za.zaak_id = zz.id
  AND zai.agenda_id = za.id
  AND (EXISTS(SELECT *
              FROM tableC zai2
                  ,tableC_type zit
              WHERE zai2.agenda_id = zai.agenda_id
                AND zai2.begindatum = zai.begindatum
                AND zit.id = zai2.item_type_id
                AND zit.naam in ('aaaa'))
  AND NOT EXISTS(SELECT *
                 FROM tableC zai2
                     ,tableC_type zit
                 WHERE zai2.agenda_id = zai.agenda_id
                   AND zai2.begindatum = zai.begindatum
                   AND zit.id = zai2.item_type_id
                   AND zit.naam in (('xxxx', 'yyyy')
  ) OR (NOT EXISTS(SELECT *
                   FROM tableC zai2
                       ,tableC_type zit
                   WHERE zai2.agenda_id = zai.agenda_id
                     AND zai2.begindatum = zai.begindatum
                     AND zit.id = zai2.item_type_id
                     AND zit.naam in ('aaaa'))
      AND EXISTS(SELECT *
                 FROM tableC zai2
                     ,tableC_type zit
                 WHERE zai2.agenda_id = zai.agenda_id
                   AND zai2.begindatum = zai.begindatum
                   AND zit.id = zai2.item_type_id
                   AND zit.naam in ('xxxx', 'yyyy')
    )))
;
```

**Comment by lukaseder** [comment, marked as answer]
You can use the online translator and pick the "jOOQ (Java)" target dialect:
https://www.jooq.org/translate/

Or use it locally via `SQLDialect.JAVA`, if you're licensed.

**Reply by nkamp** [reply]
Ah ok, I've been more than one time and I didn't understand the drop down on the right side. You have all kinds of 'output dialects' but i've not been to the end of the list! Because it is a Jooq parser I should expect than Jooq at the top of the list....
And why are all these dialects there?

I mean I'm busy with Jooq So left side SQL right side JooQ is enough? I'm thingking to simple I think, sorry. 

Best thing is, I can further with this.

**Reply by lukaseder** [reply]
> Because it is a Jooq parser I should expect than Jooq at the top of the list....

98% of the people looking for a SQL dialect translator won't look for the jOOQ dialect first...

> And why are all these dialects there?

Because it's a dialect-to-dialect translator. It says so on the website :) 

![image](https://github.com/user-attachments/assets/c98a4a64-8c25-48be-b1e4-f42470aeb09d)

Imagine migrating from Oracle to PostgreSQL

**Comment by ghost** [comment]
I prefer ChatGPT for translating SQL code into JOOQ statements. It's pretty good but I only use the 🤖  just want to save myself some writing work 😁
